module-1/src/Components/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,96 @@
+import { useMemo, useState } from "react";
+import { TestData } from "../../TestData";
+import { Genres } from "../Genres/Genres";
+import { MovieTile } from "../MovieTile/MovieTile";
+import { Search } from "../Search/Search";
+import { SortControl } from "../SortControl/SortControl";
+let sortValues = TestData.SearchFields.slice(0, 5);
+let listOfGenres = TestData.genres;
+const EMPTYSTRING = "";
+const MAINURL = "http://localhost:4000/movies?limit=20";
+
+export const MovieListPage = () => {
+  const [selectedGenre, setSelectedGenre] = useState(listOfGenres[0].genreName);
+  const [sortByField, setSortByField] = useState("title");
+  const [searchValue, setSearchValue] = useState("");
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const buildUrl = useMemo(() => {
+    let newUrl = MAINURL;
+    newUrl += `&sortBy=${sortByField}&sortOrder=asc`;
+    newUrl +=
+      searchValue === EMPTYSTRING
+        ? EMPTYSTRING
+        : `&search=${searchValue}&searchBy=title`;
+    newUrl +=
+      selectedGenre === "All" ? EMPTYSTRING : `&filter=${selectedGenre}`;
+
+    fetch(newUrl)
```

Comment:
Fetch should not be in `useMemo`. A correct but hacky approach will be to build the URL in `useMemo` and have it as a dependency for `useEffect`. That effect will call the API.

Ideally you should have a single call in an effect for the initial load. Calls for changes in sorting or filtration should be performed directly from event handlers.